src/fix/main.c

```diff
@@ -674,8 +674,8 @@ static void processFile(int input, int output, char const *name, off_t fileSize)
 		report("FATAL: Failed to read \"%s\"'s header: %s\n", name, strerror(errno));
 		return;
 	} else if (rom0Len < 0x150) {
-		report("FATAL: \"%s\" too short, expected at least 336 ($150) bytes, got only %ld\n",
-		       name, rom0Len);
+		report("FATAL: \"%s\" too short, expected at least 336 ($150) bytes, got only %jd\n",
```

Comment:
`%zd` [should work](https://stackoverflow.com/questions/586928/how-should-i-print-types-like-off-t-and-size-t) for printing `ssize_t` without a cast.

Reply:
My man pages (`system_data_types(7)`) recommends casting to `intmax_t`, which does seems to be the reasonable thing to do.

> Glibc and most other implementations provide a length modifier for `ssize_t` for the `printf(3)` and the `scanf(3)` families of functions, which is `z`; resulting commonly in `%zd` or `%zi` for printing `ssize_t` values. Although `z` works for `ssize_t` on most implementations, portable POSIX programs should avoid using it—for example, by converting the value to `intmax_t` and using its length modifier (`j`).

`size_t` must be unsigned according to POSIX, whereas `ssize_t` cannot; further, the page linked to by an answer claiming that `%zd` should work for `ssize_t` doesn't even contain `ssize_t`, so I don't think it can be trusted.